File: app/src/main/java/com/example/shgg/ApiProvider.java
package com.example.shgg;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public final class ApiProvider {

    private static Retrofit instance;
    private static String BASE_URL = "https://kr.api.riotgames.com/";

    public static Retrofit getInstance() {
        if(instance == null) {
            instance = new Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return instance;
    }
}

File: SH.GG for Kotlin/app/src/main/java/com/example/shgg/adapter/ApiProvider.kt
package com.example.shgg.adapter

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class ApiProvider {

    private lateinit var instance: Retrofit
    private var BASE_URL: String = "https://kr.api.riotgames.com/"

    fun getInstance(): Retrofit{
        instance = Retrofit.Builder()
            .baseUrl(BASE_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .build()
        return instance
    }
}